src/WebJobs.Script/Workers/Rpc/RpcWorkerChannelFactory.cs

```diff
@@ -19,24 +18,22 @@ public class RpcWorkerChannelFactory : IRpcWorkerChannelFactory
         private readonly ILoggerFactory _loggerFactory = null;
         private readonly IRpcWorkerProcessFactory _rpcWorkerProcessFactory = null;
         private readonly IScriptEventManager _eventManager = null;
-        private readonly IEnumerable<RpcWorkerConfig> _workerConfigs = null;
         private readonly IEnvironment _environment = null;
         private readonly IOptionsMonitor<ScriptApplicationHostOptions> _applicationHostOptions = null;
 
-        public RpcWorkerChannelFactory(IScriptEventManager eventManager, IEnvironment environment, IRpcServer rpcServer, ILoggerFactory loggerFactory, IOptions<LanguageWorkerOptions> languageWorkerOptions,
+        public RpcWorkerChannelFactory(IScriptEventManager eventManager, IEnvironment environment, IRpcServer rpcServer, ILoggerFactory loggerFactory, IOptionsMonitor<LanguageWorkerOptions> languageWorkerOptions,
             IOptionsMonitor<ScriptApplicationHostOptions> applicationHostOptions, IRpcWorkerProcessFactory rpcWorkerProcessManager)
         {
             _eventManager = eventManager;
             _loggerFactory = loggerFactory;
-            _workerConfigs = languageWorkerOptions.Value.WorkerConfigs;
```

Comment:
@AnatoliB / @soninaren - fix turned out to be simpler than i expected. This line is the root cause. Worker options were cached in the constructor and even though after specialization options were in fact refreshed,  creating channel was always using cached version.